content/en/blog/_posts/2025-10-23-kubelet-config-drop-in-directory-ga.md

```diff
@@ -0,0 +1,123 @@
+- **Operational overhead**: Maintaining separate, complete configuration files for each node type is error-prone and difficult to audit
+- **Change management**: Rolling out configuration changes across heterogeneous node pools requires careful coordination
+
+Previously, cluster administrators had to choose between using a single monolithic configuration file for all nodes, maintaining multiple complete configuration files, or implementing custom tooling—each approach with its own drawbacks. The Kubelet Configuration Drop-in Directory feature provides a native, built-in solution to this challenge.
+
+## What changed?
+
+With GA graduation in v1.35, the `--config-dir` flag is production-ready with stable API guarantees. This allows you to compose kubelet configuration from multiple drop-in files that are automatically merged, enabling you to maintain a base configuration while layering node-specific or environment-specific settings.
+
+## Real-world use cases
+
+### Managing heterogeneous node pools
+
+Consider a cluster with multiple node types: standard compute nodes, GPU nodes for ML workloads, and edge nodes with specialized requirements.
+
+**Base configuration** (`00-base.conf`):
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+clusterDNS:
+  - "10.96.0.10"
+clusterDomain: cluster.local
+```
+
+**GPU node override** (`50-gpu-nodes.conf`):
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+maxPods: 50
+systemReserved:
+  memory: "4Gi"
+  cpu: "1000m"
+```
+
+**Edge node override** (`50-edge-nodes.conf`):
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+evictionHard:
+  memory.available: "500Mi"
+  nodefs.available: "5%"
+```
+
+With this structure, GPU nodes apply both the base configuration and the GPU-specific overrides, while edge nodes apply the base configuration with edge-specific settings.
+
+### Gradual configuration rollouts
+
+When rolling out configuration changes, you can:
+
+1. Add a new drop-in file with a high numeric prefix (e.g., `99-new-feature.conf`)
+2. Test the changes on a subset of nodes
+3. Gradually roll out to more nodes
+4. Once stable, merge changes into the base configuration
+
+## Viewing the merged configuration
+
+Since configuration is now spread across multiple files, you can inspect the final merged configuration using the kubelet's `/configz` endpoint:
+
+```bash
+# Start kubectl proxy
+kubectl proxy
+
+# In another terminal, fetch the merged configuration
+curl -X GET http://127.0.0.1:8001/api/v1/nodes/<node-name>/proxy/configz | jq .
+```
+
+This shows the actual configuration the kubelet is using after all merging has been applied.
+
+## How to use it?
+
+As this feature has graduated to GA, there's no need to enable a feature gate. Simply make sure you are running Kubernetes v1.35 or later.
```

Comment:
specifically kubelet version is important
also: to use the feature they need to set `--config-dir` the implication by this sentence is the feature is enabled OOTB, but using the `--config-dir` flag is actually what enables it (as well as populating that directory)